lib/mail.ex

```diff
@@ -312,6 +312,30 @@ defmodule Mail do
     renderer.render(message)
   end
 
+  @doc """
+  Get messages containing attachments
+
+  Argument must be a `Mail.Message` or `List` of `Mail.Message`
+
+  Return a `List` of `Mail.Message` or emty `List`
+  """
+  def get_attachments(parts, attachments \\ [])
```

Comment:
I don't believe this is the best implementation. If we wanted to simply collect all parts that had attachments we can do this in a single function:

```elixir
def get_attachments(%Mail.Message{parts: parts}) do
  Enum.reduce(parts, [], fn(part, attachments) ->
    case Mail.Message.is_attachment?(part) do
      true -> List.insert_at(attachments, -1, part)
      false -> attachments
    end
  end)
end
```

However, I don't think this actually covers all use cases. For example, you can technically have a single-part message that is just an attachment. Also, should we return a list of the attachments themselves or the parts? Your implementation is returning the parts. I would suspect the expectation would be to return the actual attachments.

With that in mind I'd like to close this PR. I have a sense on the implementation and can probably get this done int he next 24 hours. Is that OK?